bin/aptos

```diff
@@ -22,46 +26,65 @@ const getOS = () => {
   }
 };
 
+// Only works on Unix systems. This is fine because we only need to check for brew on
+// MacOS.
+const executableIsAvailable = (name) => {
+  try {
+    execSyncShell(`which ${name}`, { encoding: "utf8" });
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
 const main = async () => {
   const os = getOS();
   const binaryName = `${PNAME}-${VERSION}-${os}`;
 
-  // If binary does not exist, download it
-  if (!fs.existsSync(`${__dirname}/${binaryName}`)) {
-    await new Promise((resolve, reject) => {
-      const url = `https://github.com/aptos-labs/aptos-core/releases/download/${PNAME}-v${VERSION}/${PNAME}-${VERSION}-${os}-x86_64.zip`;
-      console.log(`Downloading aptos CLI version ${VERSION}`);
-      let downloadProcess;
-      if (os === "Windows") {
-        downloadProcess = spawn(
-          `curl -L -o C:/tmp/aptos.zip ${url} & powershell Expand-Archive -Path C:/tmp/aptos.zip -DestinationPath C:/tmp -Force & move C:\\tmp\\aptos.exe ${__dirname}/${binaryName}`,
-          {
-            stdio: "inherit",
-            shell: true,
-          }
-        );
-      } else {
-        downloadProcess = spawn(
-          `curl -L -o /tmp/aptos.zip ${url}; unzip -o -q /tmp/aptos.zip -d /tmp; mv /tmp/aptos ${__dirname}/${binaryName};`,
-          {
-            stdio: "inherit",
-            shell: true,
-          }
-        );
-      }
+  let path;
+  if (os === "MacOS") {
+    // Confirm brew is installed.
+    const brewInstalled = executableIsAvailable("brew");
+    if (!brewInstalled) {
+      throw "Please install brew to continue: https://brew.sh/";
+    }
+    try {
+      // Get the path of where the Aptos stuff is installed, if any.
+      const directory = execSyncShell("brew --prefix aptos", { encoding: "utf8" }).toString().trim();
+      path = `${directory}/bin/aptos`;
+    } catch (e) {
+      path = "";
+    }
+  } else {
+    path = `${__dirname}/${binaryName}`;
+  }
 
-      downloadProcess.on("close", (code) => {
-        if (code !== 0) {
-          reject("Error occurred while downloading aptos CLI");
-        }
-        console.log("Download complete");
-        resolve();
-      });
-    });
+  // If binary does not exist, download it.
+  if (!fs.existsSync(path)) {
+    const url = `https://github.com/aptos-labs/aptos-core/releases/download/${PNAME}-v${VERSION}/${PNAME}-${VERSION}-${os}-x86_64.zip`;
+    let path;
+    if (os === "Windows") {
+      console.log(`Downloading aptos CLI version ${VERSION}`);
+      execSync(
+        `curl -L -o C:/tmp/aptos.zip ${url} & powershell Expand-Archive -Path C:/tmp/aptos.zip -DestinationPath C:/tmp -Force & move C:\\tmp\\aptos.exe ${path}`,
+      );
+    } else if (os === "MacOS") {
+      console.log(`Downloading aptos CLI with brew`);
```

Comment:
lets include the version we download?

Reply:
We don't really know until we download it :/

I could get the version by calling `brew info` or something in advance if you want.

Reply:
That could be useful, but I wouldn't say it's a blocker.